src/Agent/NewRelic/Agent/Extensions/NewRelic.Agent.Extensions/AwsSdk/KinesisHelper.cs

```diff
@@ -0,0 +1,103 @@
+// Copyright 2020 New Relic, Inc. All rights reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+using System;
+using System.Collections.Concurrent;
+using System.Collections.Generic;
+using System.Linq;
+
+namespace NewRelic.Agent.Extensions.AwsSdk
+{
+    public static class KinesisHelper
+    {
+
+        private static readonly ConcurrentDictionary<string, string> _streamNameCache = new();
+        private static readonly ConcurrentDictionary<Type, List<string>> _propertyInfoCache = new();
+
+        public static string GetStreamNameFromRequest(dynamic request)
+        {
+            try
+            {
+                var streamName = GetPropertyFromDynamic(request, "StreamName");
+                if (streamName != null)
+                {
+                    return streamName;
+                }
+                // if StreamName is null/unavailable, StreamARN may exist
+                var streamARN = GetStreamArnFromRequest(request);
+                if (streamARN != null)
+                {
+                    return GetStreamNameFromArn(streamARN);
+                }
+            }
+            catch
+            {
+            }
+            return null;
+        }
+
+
+        public static string GetDeliveryStreamNameFromRequest(dynamic request)
+        {
+            try
+            {
+                var streamName = GetPropertyFromDynamic(request, "DeliveryStreamName");
+                if (streamName != null)
+                {
+                    return streamName;
+                }
+                // if StreamName is null/unavailable, StreamARN may exist
+                var streamARN = GetDeliveryStreamArnFromRequest(request);
+                if (streamARN != null)
+                {
+                    return GetStreamNameFromArn(streamARN);
+                }
+            }
+            catch
+            {
+            }
```

Comment:
This is another path that could use a unit test, similar to the catch block in the other method.

Reply:
Resolved by removing unnecessary catch block